Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -109,18 +135,82 @@ extension Message {
   ) throws {
     if !data.isEmpty {
       try data.withUnsafeBytes { (body: UnsafeRawBufferPointer) in
-        if let baseAddress = body.baseAddress, body.count > 0 {
-          let pointer = baseAddress.assumingMemoryBound(to: UInt8.self)
-          var decoder = BinaryDecoder(forReadingFrom: pointer,
-                                      count: body.count,
-                                      options: options,
-                                      extensions: extensions)
-          try decoder.decodeFullMessage(message: &self)
-        }
+        try self.merge(body: body, extensions: extensions, options: options)
       }
     }
     if !partial && !isInitialized {
```

Comment:
Aren't we checking twice here also?  I guess here you need it just in an `else` clause for when data was empty.